providers.go

```diff
@@ -72,6 +72,50 @@ func (e *StdoutLogger) Warn(_ context.Context, msg string) {
 	log.Println(msg)
 }
 
+type StdoutLogLevel int
+
+const (
+	// StdoutLogDefault is equal to default value of StdoutLogLevel.
+	// Acts like StdoutLogInfo.
+	StdoutLogDefault StdoutLogLevel = iota
+	StdoutLogDebug
+	StdoutLogInfo
+	StdoutLogWarn
+	StdoutLogError
+)
+
+type StdoutLoggerf struct {
```

Comment:
Now it will look strange, since we automatically turn it into loggerf, perhaps we should leave the old signatures, just relatively modify them? Without creating new structures?

Reply:
The old logger `StdoutLogger` is quite inefficient, so I purposely created a new type and supported the logLevel feature only for the new type. It forces users to use `StdoutLoggerf` rather than `StdoutLogger`.

I want to delete the `type StdoutLogger` in the future because it does many unnecessary things (allocations, reflection, and sterilization when it is not required).

`type legacyLoggerProxy` doesn't help here at all.